crates/optimism/rpc/src/sequencer.rs

```diff
@@ -51,74 +51,50 @@ impl SequencerClient {
         self.inner.id.fetch_add(1, atomic::Ordering::SeqCst)
     }
 
-    /// Forwards a transaction to the sequencer endpoint.
-    pub async fn forward_raw_transaction(&self, tx: &[u8]) -> Result<(), SequencerClientError> {
-        let body = serde_json::to_string(&json!({
-            "jsonrpc": "2.0",
-            "method": "eth_sendRawTransaction",
-            "params": [format!("0x{}", hex::encode(tx))],
+    /// Helper function to get body of the request
+    fn body_rpc(&self, method: &str, params: Value) -> Value {
+        json!({
+                "jsonrpc": "2.0",
+            "method": method,
+            "params": params,
             "id": self.next_request_id()
-        }))
-        .map_err(|_| {
-            warn!(
-                target = "rpc::eth",
-                "Failed to serialize transaction for forwarding to sequencer"
-            );
-            SequencerClientError::InvalidSequencerTransaction
-        })?;
+        })
+    }
 
+    /// Sends a POST request to the sequencer endpoint.
+    async fn post_request(
+        &self,
+        body: String,
+        error_message: &str,
+    ) -> Result<(), SequencerClientError> {
         self.http_client()
             .post(self.endpoint())
             .header(reqwest::header::CONTENT_TYPE, "application/json")
             .body(body)
             .send()
             .await
-            .inspect_err(|err| {
-                warn!(
-                    target = "rpc::eth",
-                    %err,
-                    "Failed to forward transaction to sequencer",
-                );
+            .inspect_err(|_| {
+                warn!("{}", error_message);
             })?;
-
         Ok(())
```

Comment:
this should return the result directly without handling `?` so that the caller can inspect the error and handle it